powermeter/ema.py

```
@@ -0,0 +1,119 @@
+            ]
+
+    def _update_loop(self) -> None:
+        """Background thread: sample the underlying powermeter every ema_interval seconds."""
+        while True:
+            start = time.monotonic()
+            try:
+                raw_values = self.wrapped_powermeter.get_powermeter_watts()
+                with self._lock:
+                    self._apply_ema(raw_values)
+                self._first_reading_event.set()
+            except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)